### PR TITLE
Implementation of  Demo User

### PR DESCRIPTION
**Pull request by BerryGIT-ME**
### What this PR does (required):
- Adds the UI and front end logic to log in a user for demonstration purposes

### Screenshots / Videos (required):

demo login

demo login

#### Server
- update the MONGO_URI environment variable to point to a new database
- seed the new database by running `npm run seed`
- start the server by running `npm run dev`

#### Client
- start the client by running `npm start`
- Click the `login as a visitor button` and you will be logged in as a demo user

### Any issues with the current functionality (optional):
- The errors middleware function on the back-end had to be modified so that the JSON data sent when an error occurs matches what the client is expecting

### CODE REVIEW

File: client/src/components/DemoUser/DemoUser.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        updateSnackBarMessage('An unexpected error occurred. Please try again');
+      }
+    });
+  };
+  return (
+    <Grid container justify="center" className={classes.addMargin}>
+      <Typography className={classes.demoText}>Want to try it out first?</Typography>
+      <Button
+        onClick={() => handleLogin(EMAIL, PASSWORD)}
```
Comment (ajaytomgeorge) [inline client/src/components/DemoUser/DemoUser.tsx:35]: Is the guest user also submitting a username and password? If guest maybe you don't need them :-)

**Review by ajaytomgeorge — COMMENTED** [review]
Good work, 👍

File: client/src/components/DemoUser/DemoUser.tsx (hunk 1 above)
Comment (BerryGIT-ME) [reply client/src/components/DemoUser/DemoUser.tsx:35]: No the guest is not expected to submit a username and password. However, according to the [specifications](https://github.com/hatchways/team-brioche/issues/18#issue-992729568), when they click on the `login as a visitor` button,

File: client/src/App.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import './App.css';
 function App(): JSX.Element {
   return (
     <MuiThemeProvider theme={theme}>
+      <CssBaseline />
```
Comment (rajivtitus) [inline client/src/App.tsx:17]: 👍🏽

File: client/src/components/DemoUser/DemoUser.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import login from '../../helpers/APICalls/login';
+import useStyles from './useStyles';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import { useAuth } from '../../context/useAuthContext';
+
+// Login credentials for a demo user
+const EMAIL = 'd1@g.com';
+const PASSWORD = '123456';
+
+export default function DemoUser(): JSX.Element {
```
Comment (rajivtitus) [inline client/src/components/DemoUser/DemoUser.tsx:12]: Good thinking componentizing this! 💯

File: client/src/helpers/APICalls/bookingService.ts [hunk 4]
```diff
@@ -1,34 +1,51 @@
 import { BookingStatusType } from '../../context/useRequestContext';
 import { BookingApiData, BookingRequest } from '../../interface/BookingApiData';
-import { bookings } from '../../mocks/mockBookings';
+import { FetchOptions } from '../../interface/FetchOptions';
 
-export const getBookings = (): Promise<Array<BookingRequest>> => {
-  return Promise.resolve(bookings);
+interface UpdateBooking {
```
Comment (rajivtitus) [inline client/src/helpers/APICalls/bookingService.ts:5]: In the future please refrain from making changes in other features as the booking helper functions are not a part of this PR. You can create a small PR at any time to refactor existing code :)

**Review by rajivtitus — APPROVED** [review]
Great work with this! 👍🏽 

Minor UI suggestions: 
- Please vertically align the demo login button with the text beside it
- Maybe increase the size of the visitor login button so that it's not too small compared to the Login button?

File: client/src/components/DemoUser/DemoUser.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <Grid container justify="center" className={classes.addMargin}>
+      <Typography className={classes.demoText}>Want to try it out first?</Typography>
+      <Button
+        onClick={() => handleLogin(EMAIL, PASSWORD)}
+        variant="contained"
+        color="primary"
+        className={classes.demoButton}
+      >
+        {isSubmitting ? <CircularProgress size={18} className={classes.progress} /> : 'Login as a visitor'}
```
Comment (rajivtitus) [inline client/src/components/DemoUser/DemoUser.tsx:43]: You can call this 'Visitor Login' / 'Guest Login' instead :)

**Comment by BerryGIT-ME** [comment]
Applied all the recommended changes.

demoLogin

demosignup

**Pull request merged**